doc/guide/admin_slurm.rst

```diff
@@ -63,39 +91,46 @@ Flux has a solid security design
   :ref:`background_security`.
 
 Flux doesn't need a step manager
-  Each batch job or allocation is a new (single user) instance of Flux
-  with full capability and configurability, and a :term:`step` is really
-  a :term:`job` in Flux.  Limitations of Slurm's step manager have been
-  a source of long standing problems.
+  In Slurm, a *job* is a resource allocation scheduled by the Slurm scheduler.
+  Within the allocation, a *job step* is a unit of work scheduled by the the
+  Slurm step manager. Since a new Flux instance can be launched as a
+  :term:`job` in another Flux instance (subdividing its resources), there is
+  no need have a step abstraction or a step manager.  All units of work are
+  jobs scheduled and launched by the same robust system.  Limitations of
+  Slurm's step manager have been a source of long standing problems.
 
 Flux APIs are not an afterthought
   Well thought out Python and C APIs make it easy to integrate Flux with
   workflow systems and new environments.
 
 Flux has a rich resource representation
   Flux's design incorporates a graph-based resource model that is
-  extensible to arbitrary resource types.  Caveat: not yet fully implemented.
+  extensible to arbitrary resource types.
 
 Flux is dynamic
-  Flux supports growing and shrinking allocations.  Caveat: not yet fully
-  implemented.
+  The Flux design supports growing and shrinking allocations.
 
 Flux is scalable
   Flux's recursive launch design enables each allocation to scale
   independently.  Each instance of Flux has job throughput and node count
   scalability comparable to Slurm.  But a cluster will typically be running
   many instances of Flux compared to one instance of Slurm.
```

Comment:
If I can get flux-hierarchy optimized, we can at least report a throughput in the thousands. Math says we should get much higher than that, but math is limited by me. :)

Reply:
We have the classic flux-tree, which did the same thing, but this already mentions the recursive parallelism so I would call it covered personally.

Reply:
flux-tree was removed (we have it still in tutorial content if you need me to dig it up), and (if you haven't looked at the code recently) it's a hugely complex spaghetti bash script with a Python wrapper to plug into the flux cli that (imho) would be unmaintainable. I have a nice library that is going to make this a more production setup, also affording more advanced use cases like creating the hierarchy of brokers to later submit to with different communication strategies. For example, `flux-tree` won't work to provide a service of flux hierarchies that exist across centers.

Reply:
Here is flux-tree:

- https://github.com/flux-framework/Tutorials/blob/ef36e935ba8665d1b3ea61d567918d779953bfe2/2025/ISC-AWS/JupyterNotebook/flux-tree/flux-tree#L123. it's great for a demo, but doesn't fit the bill for the kind of interactions I want, and that will be reflected in future cases for multi-cluster scheduling.
- I am working on https://github.com/converged-computing/flux-hierarchy/ as a more production tool. We already have a setup that can handle a URI interface, interfaces to start / stop / submit / measure throughput, and node executors for different node or cluster types.